File: source/algorithm/NQ026/NQ026_improved.cpp
#include <cmath>
#include <iostream>
using namespace std;

int main() {
    int t; cin >> t;
    while (t--) {
        double s, u; scanf("%lf%lf", &s, &u);
        double v = 1.0 / tan(atan(1.0/s) - atan(1.0/u));  // 计算v值
        double res = v*u - s*u - v*s;  // 计算最终结果
        printf("%.0lf\n", res);
    }
    return 0;
}

File: 兴趣班入门百练讲义/NQ100/NQ026/NQ026_improved.py
import math

def main():
    t = int(input())
    for _ in range(t):
        s, u = map(float, input().split())
        v = 1.0 / math.tan(math.atan(1.0/s) - math.atan(1.0/u))  # 计算v值
        res = v*u - s*u - v*s  # 计算最终结果
        print("{0:.0f}".format(res))

if __name__ == "__main__":
    main()